internal/webhook/v1alpha1/bmcsettings_webhook_test.go

```diff
@@ -0,0 +1,148 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and IronCore contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	metalv1alpha1 "github.com/ironcore-dev/metal-operator/api/v1alpha1"
+	. "sigs.k8s.io/controller-runtime/pkg/envtest/komega"
+)
+
+var _ = Describe("BMCSettings Webhook", func() {
+	var (
+		BMCSettingsV1 *metalv1alpha1.BMCSettings
+		validator     BMCSettingsCustomValidator
+	)
+
+	BeforeEach(func() {
+		BMCSettingsV1 = &metalv1alpha1.BMCSettings{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace:    "ns.Name",
+				GenerateName: "test-",
+			},
+			Spec: metalv1alpha1.BMCSettingsSpec{
+				Version:                 "P70 v1.45 (12/06/2017)",
+				SettingsMap:             map[string]string{},
+				BMCRef:                  &v1.LocalObjectReference{Name: "foo"},
+				ServerMaintenancePolicy: metalv1alpha1.ServerMaintenancePolicyEnforced,
+			},
+		}
+		By("Creating an BMCSettings")
+		Expect(k8sClient.Create(ctx, BMCSettingsV1)).To(Succeed())
+		DeferCleanup(k8sClient.Delete, BMCSettingsV1)
+		validator = BMCSettingsCustomValidator{Client: k8sClient}
+		SetClient(k8sClient)
+
+	})
+
+	AfterEach(func() {
+		// TODO (user): Add any teardown logic common to all tests
+	})
+
+	Context("When creating or updating BMCSettings under Validating Webhook", func() {
+
+		It("Should deny creation if a BMC referred is already referred by another", func(ctx SpecContext) {
+			By("Creating another BMCSettings with reference to existing referred BMC")
+			BMCSettingsV2 := &metalv1alpha1.BMCSettings{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace:    "ns.Name",
+					GenerateName: "test-bmc-",
+				},
+				Spec: metalv1alpha1.BMCSettingsSpec{
+					Version:                 "1.45.455b66-rev4",
+					SettingsMap:             map[string]string{},
+					BMCRef:                  &v1.LocalObjectReference{Name: "foo"},
+					ServerMaintenancePolicy: metalv1alpha1.ServerMaintenancePolicyEnforced,
+				},
+			}
+			Expect(validator.ValidateCreate(ctx, BMCSettingsV2)).Error().To(HaveOccurred())
```

Comment:
iirc, `Expect(...).To(HaveOccurred())` should work without the `Error()` in between.